livesync_citadel/indico_livesync_citadel/schemas.py

```diff
@@ -0,0 +1,205 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2020 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from marshmallow import post_dump
+from tika import parser
+from webargs import fields
+
+from indico.core.db.sqlalchemy.links import LinkType
+from indico.core.marshmallow import mm
+from indico.modules.attachments.models.attachments import Attachment, AttachmentType
+from indico.modules.events import Event
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.notes.models.notes import EventNote
+from indico.web.flask.util import url_for
+
+
+class ACLSchema(object):
+    _access = fields.Method('_get_default_acl')
+
+    def _get_default_acl(self, *args):
+        """Return the object ACLs. More information here https://cern-search.docs.cern.ch/usage/permissions/"""
+        from indico_livesync_citadel.plugin import LiveSyncCitadelPlugin
+        default_acl =  LiveSyncCitadelPlugin.settings.get('search_owner_role')
+        return {
+            'owner': [default_acl],
+            'read': [default_acl],
+            'update': [default_acl],
+            'delete': [default_acl]
+        }
+
+
+def _get_location(obj):
+    if obj.venue_name and obj.room_name:
+        return '{}: {}'.format(obj.venue_name, obj.room_name)
+    elif obj.venue_name or obj.room_name:
+        return obj.venue_name or obj.room_name
+    else:
+        return ''
+
+
+def _get_category_path(obj):
+    if isinstance(obj, Event):
+        event = obj
+    elif isinstance(obj, Attachment):
+        event = obj.folder.event
+    else:
+        event = obj.event
+    return event.category.chain_titles
+
+
+def _get_people_list(obj):
+    return [
+        '{} ({})'.format(pl.full_name, pl.affiliation) if pl.affiliation else pl.full_name
+        for pl in obj.person_links
+    ]
+
+
+class _EventDataSchema(mm.ModelSchema):
+    id = fields.Int()
+    category_path = fields.Function(_get_category_path)
+    event_type = fields.Str()
+    title = fields.Str()
+    description = fields.Str()
+    location = fields.Function(_get_location)
+    speakers_chairs = fields.Function(_get_people_list)
+    url = fields.Str(attribute='external_url')
+
+    class Meta:
+        model = Event
+        fields = ('id', 'category_path', 'event_type', 'title', 'description', 'location', 'speakers_chairs', 'url')
+
+
+class EventSchema(mm.Schema, ACLSchema):
```

Comment:
Wouldn't this inheritance order make more sense? that way we first inherit from mm.Schema and then add the mixin (which should, btw, be named `ACLSchemaMixin`)

```suggestion
class EventSchema(ACLSchema, mm.Schema):
```